### PR TITLE
Weather card: Add wind speed direction

### LINKED ISSUE #5924: Weather card: Wind Speed has no direction attribute
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem
Wind speed as an attribute is missing direction even if wind speed includes direction in the weather attributes.

## Expected behavior
It should show speed and direction, also word "speed" seems redundant.

## Environment

- Home Assistant release with the issue: 0.110.0b3
- Last working Home Assistant release (if known): 0.108.x
- Browser and browser version: Chrome 81.0.4044
- Operating system: Android 10

## State of relevant entities

```yaml

```

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

```txt

```

## Additional information
Screenshot_20200518-213851_Chrome
Screenshot_20200518-213533_Chrome

### PR DESCRIPTION
**Pull request by danimart1991**
## Proposed change
- Add wind direction to Weather Card. Currently only Wind Speed is showed.
- Improved the code moving repeated code to weather.ts

image

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #5924 

## Checklist

- [X] The code change is tested and works locally.
- [X]  There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @danimart1991, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#7202).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: src/panels/lovelace/cards/hui-weather-forecast-card.ts [hunk 1]
```diff
@@ -224,13 +225,21 @@ class HuiWeatherForecastCard extends LitElement implements LovelaceCard {
...
+                            ${getWeatherUnit(
+                              this.hass,
+                              this._config.secondary_info_attribute
+                            )}
+                          `
+                        : getWind(
+                            this.hass,
+                            stateObj.attributes.wind_speed,
+                            stateObj.attributes.wind_bearing
+                          )}
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-weather-forecast-card.ts:242]: I think this should be reversed. If it is wind speed do something different else case for everything else

**Comment by zsarnett** [comment]
Make sure to add windspeed to the weather entity type

File: src/panels/lovelace/cards/hui-weather-forecast-card.ts (hunk 1 above)
Comment (danimart1991) [reply src/panels/lovelace/cards/hui-weather-forecast-card.ts]: You are right. Changed

**Comment by danimart1991** [comment]
> Make sure to add windspeed to the weather entity type

Sorry. First PR in HA Frontend.

**Comment by zsarnett** [comment]
All good. This looks great!

File: src/data/weather.ts [hunk 2]
```diff
@@ -77,13 +77,29 @@ const getWindBearingText = (degree: string): string => {
...
+): string => {
+  if (bearing != null) {
+    const cardinalDirection = getWindBearing(bearing);
+    return `${speed} ${getWeatherUnit(hass!, "length")}/h (${
+      hass.localize(
+        `ui.card.weather.cardinal_direction.${cardinalDirection.toLowerCase()}`
+      ) || cardinalDirection
+    })`;
+  }
+  return `${speed} ${getWeatherUnit(hass!, "length")}/h`;
```
Comment (zsarnett) [inline src/data/weather.ts:100]: Should the `/h` be here? it's in the getWeatherUnit function right?

**Comment by zsarnett** [comment]
Also, just so we know its all the same. Can you send a screenshot of the More info dialog?

**Comment by danimart1991** [comment]
> Also, just so we know its all the same. Can you send a screenshot of the More info dialog?

Yes of course.

metric:
image

imperial:
image

File: src/data/weather.ts (hunk 2 above)
Comment (danimart1991) [reply src/data/weather.ts:100]: It seems weird to mee too, but I don't modify this methods I only move them to weather.ts.

Comment (zsarnett) [reply src/data/weather.ts:100]: hmm its because we are using the length unit here. should it be windspeed?

Comment (danimart1991) [reply src/data/weather.ts:100]: You are right. Just below in the getWeatherUnit, the wind_speed case do the same. I change it to wind_speed and remove the /h.

**Review by zsarnett — APPROVED** [review]

**Comment by zsarnett** [comment]
Looks Great! Thank you for your Contribution!

**Comment by zsarnett** [comment]
🐬 🐬 🐬 🐬

File: src/data/weather.ts [hunk 3]
```diff
@@ -77,13 +77,29 @@ const getWindBearingText = (degree: string): string => {
...
   }
   return "";
 };
 
+export const getWind = (
+  hass: HomeAssistant,
+  speed: string,
+  bearing: string
+): string => {
+  if (bearing != null) {
```
Comment (zsarnett) [inline src/data/weather.ts:92]:
```suggestion
  if (bearing !== null) {
```

Found one little thing here which I think may have been in the old code. Bram will review this as well and merge if good :)

Comment (danimart1991) [reply src/data/weather.ts:92]: looks good to me.

**Comment by kukulich** [comment]
Is it possible to make it translatable ?

**Comment by danimart1991** [comment]
> Is it possible to make it translatable ?

The "Wind Speed" string is already translatable. 🤔

**Comment by kukulich** [comment]
@danimart1991 I mean "WNW" etc.

**Comment by danimart1991** [comment]
> @danimart1991 I mean "WNW" etc.

It's hardcoded. Maybe the best option is to add another issue. I don't even know how to do it.

**Comment by kukulich** [comment]
@danimart1991 Nevermind, it looks it's already translated: https://github.com/home-assistant/frontend/pull/7202/files#diff-b686dd2985c68dcf3742ee89f46d44bcR95

**Review by bramkragten — APPROVED** [review]

**Pull request merged**